src/main/java/org/cyclonedx/gradle/utils/ExternalResourcesUtil.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * This file is part of CycloneDX Gradle Plugin.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) OWASP Foundation. All Rights Reserved.
+ */
+package org.cyclonedx.gradle.utils;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import org.cyclonedx.model.ExternalReference;
+
+public class ExternalResourcesUtil {
```

Comment:
This class requires benchmarking. I afraid it can significantly increase BOM generation time. You can use gradle profiler for it https://github.com/gradle/gradle-profiler

Reply:
@skhokhlov 
On a large project, I have run each case and named the file accordingly. 
It seems, that adding this feature does not have a large impact on performance.
I also checked each generated sbom and each case works fine. the change can successfully pull the remote url either from custom configuration, from environment variables (e.g. Jenkins) and from the git repo config. In this order.

For completeness sake, I have also attached the scenarios file.
[performance.scenarios.txt](https://github.com/user-attachments/files/17957833/performance.scenarios.txt)

[benchmark-v1.10.0.csv](https://github.com/user-attachments/files/17957842/benchmark-v1.10.0.csv)
[benchmark-custom-configuration.csv](https://github.com/user-attachments/files/17957848/benchmark-custom-configuration.csv)
[benchmark-from-env-var.csv](https://github.com/user-attachments/files/17957851/benchmark-from-env-var.csv)
[benchmark-from-repo-config.csv](https://github.com/user-attachments/files/17957855/benchmark-from-repo-config.csv)